### DISCUSSION TITLE
VisualStudioCode and WSL on a WIN10. Can't compile a Vyper smart contract...

### DISCUSSION THREAD

**Discussion by conceptstom** [vyperlang/vyper#2365] [General]
I can't compile a simple smart contract(screenshot) in Vyper in VisualStudioCode(VSC) with a Vyper Extension. I run it in a WSL(WinSubsystemLinux) - hence this terminal in VSC. Pythons is installed properly. The Remote Extension for VSC is activated also...Can't find Vyper: Command not found... anybody also uses Vyper in VSC ?
![Skærmbillede 2021-05-07 04 38 16](https://user-images.githubusercontent.com/8062795/117517962-38594480-af9e-11eb-9c23-e1f3636fd578.png)

**Comment by fubuloubu** [comment]
@sabotagebeats might have some more insight

**Comment by notaquant** [comment]
Assuming you're running under Remote WSL (i.e opening directories using WSL paths and configuration) you maybe need to add `vyper` to your bash .profile file.

On my machine `whereis vyper` yields the following 

```sh
vyper: /home/user/.local/bin/vyper
```
```
# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/.local/bin" ] ; then
    PATH="$HOME/.local/bin:$PATH"
fi
```

**Reply by conceptstom** [reply]
Thanks for your reply!
Yes it's exactly the same path for the vyper as you have. I can see in the path and if i use : Which Vyper. I've also put in the Vyper field under SETTINGS in the installed Vyper extension in VisualStudioCode... But i stiil get the error as seen on the screenshot.
![Skærmbillede 2021-05-13 21 00 24](https://user-images.githubusercontent.com/8062795/118174845-9fa84600-b42f-11eb-90b3-0ef5378d76db.png)

**Comment by jondoescoding** [comment]
> Assuming you're running under Remote WSL (i.e opening directories using WSL paths and configuration) you maybe need to add `vyper` to your bash .profile file.
> 
> On my machine `whereis vyper` yields the following
> 
> ```shell
> vyper: /home/user/.local/bin/vyper
> ```
> 
> ```
> # set PATH so it includes user's private bin if it exists
> if [ -d "$HOME/.local/bin" ] ; then
>     PATH="$HOME/.local/bin:$PATH"
> fi
> ```



Having the same issue on my PC as well.

OS: Win10
IDE: VSCode (running a remote instance of WSL)

I have a .bash_profile inside of my directory could you explain how I'd set it up there using this method

**Reply by jondoescoding** [reply]
Figured it out. 

Steps
- I installed vyper via the snap available on the github
- Type whereis vyper in the terminal
- Whatever output I got I transfered it into my settings and place it into the empty box that is labelled "vyper command"
